framework/src/main/java/org/checkerframework/framework/ajava/JointJavacJavaParserVisitor.java

```diff
@@ -0,0 +1,2382 @@
+        if (!(javaParserNode instanceof AssertStmt)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, AssertStmt.class);
+        }
+
+        AssertStmt node = (AssertStmt) javaParserNode;
+        processAssert(javacTree, node);
+        javacTree.getCondition().accept(this, node.getCheck());
+        ExpressionTree detail = javacTree.getDetail();
+        assert (detail != null) == node.getMessage().isPresent();
+        if (detail != null) {
+            detail.accept(this, node.getMessage().get());
+        }
+
+        return null;
+    }
+
+    @Override
+    public Void visitAssignment(AssignmentTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof AssignExpr)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, AssignExpr.class);
+        }
+
+        AssignExpr node = (AssignExpr) javaParserNode;
+        processAssignment(javacTree, node);
+        javacTree.getVariable().accept(this, node.getTarget());
+        javacTree.getExpression().accept(this, node.getValue());
+        return null;
+    }
+
+    @Override
+    public Void visitBinary(BinaryTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof BinaryExpr)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, BinaryExpr.class);
+        }
+
+        BinaryExpr node = (BinaryExpr) javaParserNode;
+        processBinary(javacTree, node);
+        javacTree.getLeftOperand().accept(this, node.getLeft());
+        javacTree.getRightOperand().accept(this, node.getRight());
+        return null;
+    }
+
+    @Override
+    public Void visitBlock(BlockTree javacTree, Node javaParserNode) {
+        if (javaParserNode instanceof InitializerDeclaration) {
+            return javacTree.accept(this, ((InitializerDeclaration) javaParserNode).getBody());
+        }
+
+        if (!(javaParserNode instanceof BlockStmt)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, BlockStmt.class);
+        }
+
+        BlockStmt node = (BlockStmt) javaParserNode;
+        processBlock(javacTree, node);
+        processStatements(javacTree.getStatements(), node.getStatements());
+        return null;
+    }
+
+    /**
+     * Given a matching sequence of statements for a block, visits each javac statement with its
+     * corresponding JavaParser statememt, excluding synthetic javac trees like no-argument
+     * constructors.
+     *
+     * @param javacStatements sequence of javac trees for statements
+     * @param javaParserStatements sequence of JavaParser statements representing the same block as
+     *     {@code javacStatements}
+     */
+    private void processStatements(
+            Iterable<? extends StatementTree> javacStatements,
+            Iterable<Statement> javaParserStatements) {
+        Iterator<? extends StatementTree> javacIter = javacStatements.iterator();
+        boolean hasNextJavac = javacIter.hasNext();
```

Comment:
Here and elsewhere, you could drop the boolean and just test whether `javacStatement` is null.  That would be a bit shorter.  But you can leave this code as is if you prefer.